### PR TITLE
Fix evaluator to use default on undefined value provided

### PR DESCRIPTION
**Pull request by Sindbag**
Price(undefined, undefined) -> Price(0, 0) as a default value when no value is provided for initialisation.
Regression tests added for evaluator.

### CODE REVIEW

File: src/blockchains/eth_generator/consts.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+}
+export interface IWideDataType extends IDTStub { [key:string]: string|number|boolean|undefined|null, value: string|number|boolean; life: number; update: number }
 export const evaluator = (d: IWideDataType) => {
     switch (d.type) {
-        case 'price': return `Price(${d.value}, ${d.decimals})`
+        case 'price': return !!d.value ? `Price(${d.value}, ${!!d.decimals ? d.decimals : 0})` : PUSH_CONSTRUCTION[d.type].default
         default: break;
     }
-    return d.value;
+    return !!d.value ? d.value : PUSH_CONSTRUCTION[d.type].default;
```
Comment (zlumer) [inline src/blockchains/eth_generator/consts.ts]: `!!a ? a : b` = `a || b`

File: src/blockchains/eth_generator/consts.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import "jest-extended"
+import {evaluator, IWideDataType, PUSH_CONSTRUCTION} from "./consts";
+
+describe("Regression test for empty value or decimals", () => {
```
Comment (zlumer) [inline src/blockchains/eth_generator/consts.test.ts]: `skip()` this tests for now

File: src/blockchains/eth_generator/codepresets.test.ts [hunk 3]
```diff
@@ -117,4 +118,19 @@ describe('ETH Contract constructor', () => {
...
     })
+
+    ;[0, undefined, null].forEach(val => {
+        let obj = <IWideDataType>{ value: 0, name: 'some', hash: 'exist', update: val, life: 1 }
+        ;['update', 'life', 'hash'].forEach(prop => {
+            it(`should fire exception on ${prop} equal to ${val}`, () => {
+                obj[prop] = val;
+                try {
+                    getContractBase('name', [obj]);
+                    expect(false).toBe(true)
```
Comment (zlumer) [inline src/blockchains/eth_generator/codepresets.test.ts]: `expect(...).toThrow()` would be more clear
https://jestjs.io/docs/en/expect.html#tothrowerror

File: src/blockchains/eth_generator/consts.ts [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
     type: ETHType
-    value: string
     decimals?: number
 }
-export interface IWideDataType extends IETHDataType { value: any; life: number; update: number }
+export interface IETHDataType extends IDTStub
+{
+    value: string
+}
+export interface IWideDataType extends IDTStub { [key:string]: string|number|boolean|undefined|null, value: string|number|boolean; life: number; update: number }
```
Comment (zlumer) [inline src/blockchains/eth_generator/consts.ts]: Code style: `interface` should be multiline

File: src/blockchains/eth_generator/consts.ts [hunk 5]
```diff
@@ -1,21 +1,24 @@
...
 export const evaluator = (d: IWideDataType) => {
     switch (d.type) {
-        case 'price': return `Price(${d.value}, ${d.decimals})`
+        case 'price': return !!d.value ? `Price(${d.value}, ${!!d.decimals ? d.decimals : 0})` : PUSH_CONSTRUCTION[d.type].default
```
Comment (zlumer) [inline src/blockchains/eth_generator/consts.ts]: Intention not clear: better use string interpolation in both cases with `d.value || PUSH_CONSTRUCTION[d.type].default`
If `d.value` was passed, `d.decimals` MUST be passed as well.

File: src/blockchains/eth_generator/consts.test.ts (hunk 2 above)
Comment (Sindbag) [reply src/blockchains/eth_generator/consts.test.ts]: The test checks, that independent constructor module uses default value as fallback and if provided uses the provided initial value. There could be another test on ducor-backend level, that takes the whole system behaviour, not just eth-generator.

**Pull request merged**